testcases/kernel/syscalls/statx/statx05.c

```diff
@@ -86,9 +87,22 @@ static void run(unsigned int i)
 
 static void setup(void)
 {
+	FILE *f;
 	char opt_bsize[32];
 	const char *const fs_opts[] = {"-O encrypt", opt_bsize, NULL};
-	int ret;
+	int ret, rc, major, minor, patch;
+
+	f = popen("mkfs.ext4 -V 2>&1 | awk '/mke2fs/ {print $2}'", "r");
```

Comment:
We have `SAFE_POPEN()` which handles the errors (thus `if (!f)` clause is not needed). It exits the test on error, but IMHO there is no reason to continue on error.

Reply:
I tried this with:
```
   #include "tst_safe_stdio.h"
...
    f = SAFE_POPEN("mkfsds.ext4 -V 2>&1", "r");
    rc = fscanf(f, "mke2fs %d.%d.%d", &major, &minor, &patch);
    if (rc != 3)
        tst_res(TWARN, "Unable parse version number");
    else if (major * 10000 + minor * 100 + patch < 14300)
        tst_brk(TCONF, "Test needs mkfs.ext4 >= 1.43 for encrypt option, test skipped");
```
But even with a non-existing command, the test won't terminate with the unsuccessful SAFE_POPEN like you mentioned here. Not sure if I am using this right.

Reply:
Looking at popen() manual non-existing command is not handled at the popen() since it executes a shell and instead you will get error at pclose().

Reply:
Yes: *The pclose() function returns -1 if wait4(2) returns an error, or some other error is detected.*
(https://linux.die.net/man/3/pclose).

But IMHO it's not needed to detect exit code of `mkfs.foo` command, *Unable parse version number* would be enough for me. I meant to use [`SAFE_POPEN()`](https://github.com/linux-test-project/ltp/blob/master/lib/safe_stdio.c#L73), because it does some other error handling.

Reply:
OK thanks for your input!